src/lib/raft/raft.h

```diff
@@ -292,6 +313,17 @@ raft_cfg_is_enabled(struct raft *raft, bool is_enabled);
 void
 raft_cfg_is_candidate(struct raft *raft, bool is_candidate);
 
+/**
+ * Notify the instance whether it has a quorum of bidirectional connections to
+ * the remote peers.
+ */
+void
+raft_notify_have_quorum(struct raft *raft, bool have_quorum);
+
+/** Notify this instance whether its peer sees the current leader. */
+void
+raft_notify_leader_seen(struct raft *raft, bool leader_seen, uint32_t source);
+
 /**
  * Bump the term and become a candidate for it regardless of the config. In case
```

Comment:
Could you please cover the lib/raft changes in raft's unit tests? So as the library would still be self-sufficient, not depending on being tested somewhere outside.

Reply:
Done.